Lesson16/Lesson16-Busra/busra.js

```diff
@@ -0,0 +1,442 @@
+    - Define a function `sumWithDoWhile(numbers)` that uses a do-while loop
+      to sum all numbers in the array.
+    - Log the total.
+*/
+const numbersEx11 = [1,2,3,4,5,6,7,8,9,10];
+
+function sumWithDoWhile(arr){
+    let i=0;
+    let sumEx11 = 0;
+  do{
+    sumEx11 += arr[i];
+    i++;
+  }
+  while(i < arr.length){
+   
+   console.log(sumEx11);
+
+  }
+ 
+}
+
+sumWithDoWhile(numbers);
+
+/*
+12. Remove Duplicates from an Array
+    - Define a function `removeDuplicates(arr)` that loops through the array
+      and creates a new array without duplicate elements.
+    - Hint: you could check if the item is already in the new array before pushing.
+    - Log the new array without duplicates.
+*/
+
+let ArrayEx12 = [1,2,3,3,4,4,5,6,7,8,8,9,10];
+
+function removeDuplicates(arr){
+  let newArray = [];
+  for(let i=0; i< arr.length; i++){
+
+    if(!newArray.includes(arr[i])){
+      newArray.push(arr[i]);
+    }
+  }
+  console.log(newArray);
+}
+
+removeDuplicates(ArrayEx12);
+
+
+/*
+13. Calculate Factorial (For Loop)
+    - Define a function `factorial(n)` that calculates n! (n factorial)
+      using a for loop.
+    - Log the result. 
+    - Example: factorial(5) -> 120
+*/
+
+function factorial(n){
+  let calculator = 1;
+  for(let i=1; i<=n; i++ ){
+     calculator = calculator * i;
+  }
+  return calculator;
+}
+factorial(5);
+
+/*
+14. String -> Array -> String
+    - Define a function `reverseWords(sentence)` that splits the sentence 
+      into an array of words, reverses the array order, then joins it back into
+      a string. Use loops or built-in methods as you like.
+    - Log the reversed sentence.
+*/
+const sentenceEx14 = "Javascript is the most fun programming language I have ever seen so far";
+
+function reverseWords(sentence) {
+  let splitSentence = sentence.split(" ");
+  let reversedSentence = [];
+  for (let i = splitSentence.length - 1; i >= 0; i--) {
+    reversedSentence.push(splitSentence[i]);
+  }
+  return reversedSentence.join(" ");
+}
+
+console.log(reverseWords(sentenceEx14));
+
+
+
+/*
+15. Filter Words Longer Than X
+    - Define a function `filterLongWords(words, minLength)` that uses a for loop
+      to collect only the words that have a length >= minLength.
+    - Log the resulting array.
+*/
+const myWords=["bir","iki","üç","dört","beş","altı","yedi","sekiz"];
+
+
+
+function filterLongWords(words, minLength){
+  let newWords = [];
+  for(let i=0; i < words.length; i++){
+    if(words[i].length >= minLength){
+      newWords.push(words[i]);
+    }
+    
+  }
+  return newWords;
+}
+
+filterLongWords(myWords, 3);
+
+
+/*
+16. Log Array Elements with Their Indices
+    - Define a function `logElementsWithIndex(arr)` that loops through the array
+      and logs "Index: i, Value: arr[i]" for each element.
+*/
+
+const myArrayEx16 = ["busra","demirkesen",34,true,false];
+function logElementsWithIndex(arr){
+  for(let i=0; i<myArrayEx16.length; i++){
+    console.log(i,myArrayEx16[i]);
```

Comment:
This statement works but try to always make your code and log messages easy to read.
```suggestion
    console.log(`Index: ${i}, Value: ${myArrayEx16[i]}.`);
```

Having your code clean reduces amount of mistakes significantly.